python/templates/macros/implementations.jinja2

```diff
@@ -19,7 +19,24 @@
 }
 
 Mutable{{ type }} {{ full_type }}::clone() const {
+{% if prefix %}
   return Mutable{{ type }}(podio::utils::MaybeSharedPtr(new {{ type }}Obj(*m_obj), podio::utils::MarkOwned));
+{% else %}
+  auto tmp = new {{ type }}Obj();
+  tmp->id = podio::ObjectID();
+  tmp->data = m_obj->data;
+{% for relation in multi_relations %}
+  // If the current object has been read from a file, then the object may only have a slice of the relation vector
+  // so this slice has to be copied in case we want to modify it
+  tmp->m_{{ relation.name }}->reserve(m_obj->m_{{ relation.name }}->size());
+  for (size_t i = m_obj->data.{{ relation.name }}_begin; i < m_obj->data.{{ relation.name }}_end; i++) {
+    tmp->m_{{ relation.name }}->push_back((*m_obj->m_{{ relation.name }})[i]);
```

Comment:
```suggestion
    tmp->m_{{ relation.name }}->emplace_back((*m_obj->m_{{ relation.name }})[i]);
```
Might be able to skip one of the checks in `MabyeSharedPtr` if we get a move this way.

Reply:
I'm not sure this can be a move, `[]` should return something movable no? In any case you don't want to move since you don't want to lose the original relation after a `clone()`

Reply:
You are probably right, we won't get a move, but we should save creating a temporary for the `push_back`, and instead just get one copy constructor call.